Framework/Kernel/inc/MantidKernel/NexusDescriptor.h

```diff
@@ -90,6 +90,11 @@ class MANTID_KERNEL_DLL NexusDescriptor {
   std::string pathOfType(const std::string &type) const;
   /// Query if a given type exists somewhere in the file
   bool classTypeExists(const std::string &classType) const;
+  // Recursively search for the first group which contains a class of type
+  // classType
+  static bool findAndOpenParentGroup(::NeXus::File &file,
```

Comment:
The name of the function doesn't quite fit the description. Would imply opening the parent of some current with existing naming. `findAndOpenFirstChild`?